specs/Workers.md

```diff
@@ -0,0 +1,1402 @@
+  /// worker is executing. See the [MDN documentation]
+  /// (https://developer.mozilla.org/docs/Web/API/Worker/scriptURL) for more information.
+  ///
+  /// The caller must free the returned string with `CoTaskMemFree`.  See
+  /// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings).
+  [propget] HRESULT ScriptUri([out, retval] LPWSTR* value);
+
+  /// Add an event handler for the `Destroyed` event that is raised when the
+  /// worker object is destroyed.
+  ///
+  /// A worker object is destroyed when the worker script is terminated or when
+  /// the `CoreWebView2DedicatedWorker` object is destroyed.
+  HRESULT add_Destroyed(
+      [in] ICoreWebView2DedicatedWorkerDestroyedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Removes an event handler previously added with `add_Destroyed`.
+  HRESULT remove_Destroyed(
+      [in] EventRegistrationToken token);
+}
+
+
+[uuid(ad6921d4-c416-5945-8437-2c97aeb76e6e), object, pointer_default(unique)]
+interface ICoreWebView2Profile2 : IUnknown {
+  /// Get the service worker manager to monitor service worker registrations and
+  /// interact with the service worker associated with the current profile.
+  ///
+  /// The changes would apply to the context of the user profile. That is,
+  /// other WebViews under the same user profile could be affected.
+  ///
+  /// \snippet ScenarioServiceWorkerManager.cpp ServiceWorkerManager
+  [propget] HRESULT ServiceWorkerManager([out, retval] ICoreWebView2ServiceWorkerManager** value);
+
+
+  /// Get the shared worker manager to monitor shared worker creations and interact
+  /// with the shared worker associated with the current profile.
+  ///
+  /// The changes would apply to the context of the user profile. That is, other
+  /// WebViews under the same user profile could be affected.
+  ///
+  /// \snippet ScenarioSharedWorkerManager.cpp SharedWorkerManager
+  [propget] HRESULT SharedWorkerManager([out, retval] ICoreWebView2SharedWorkerManager** value);
+}
+
+[uuid(4e07e562-8db7-5815-907b-6d89a253a974), object, pointer_default(unique)]
+interface ICoreWebView2ServiceWorkerManager : IUnknown {
+  /// Adds an event handler for the `ServiceWorkerRegistered` event.
+  ///
+  /// A ServiceWorker is a specific type of worker that takes a JavaScript file
+  /// that can control the web-page/site that it is associated with,
+  /// intercepting and modifying navigation and resource requests, and caching
+  /// resources in a very granular fashion to give you complete control
+  /// over how app behaves in certain situations.
+  ///
+  /// Service workers essentially act as proxy servers that sit between web applications,
+  /// the browser, and the network (when available).
+  /// Unlike Shared Workers, which have their own separate global scope, Service
+  ///Workers have no DOM access and run in a different context.
```

Comment:
Confusing "Unlike..." sentence since the two parts do not contrast. Usually, "Unlike A, which does X, B does Y." you expect X and Y to be contrasting, but here the X and Y seem to talk about unrelated things.